Tests/Fixtures/Message.php

```diff
@@ -64,7 +64,11 @@ public function getHeaderLine($name)
 
     public function withHeader($name, $value)
     {
-        throw new \BadMethodCallException('Not implemented.');
+        if (is_string($value) || is_array($value)) {
+            $this->headers[$name] = $value;
```

Comment:
wrong implementation. If the value is a string, you need to wrap it in an array. Otherwise, ``getHeader`` will not return an array